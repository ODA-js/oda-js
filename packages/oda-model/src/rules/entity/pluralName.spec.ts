import Rule from './pluralName';

describe('pluralName', () => {
  let rule: Rule;
  beforeAll(() => {
    rule = new Rule();
  });

  it('throw', () => {

    const result = rule.validate({
      entity: {
        plural: 'species',
        name: 'species',
      },
    } as any);

    expect(result).toMatchSnapshot();

  });

  it('not throw', () => {
    const result = rule.validate({
      entity: {
        plural: 'AllSpecies',
        name: 'species',
      },
    } as any);

    expect(result).toMatchSnapshot();

  });
});
