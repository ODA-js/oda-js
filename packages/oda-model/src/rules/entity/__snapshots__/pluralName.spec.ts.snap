// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pluralName not throw 1`] = `Array []`;

exports[`pluralName throw 1`] = `
Array [
  Object {
    "entity": "species",
    "message": "plural form of entity\`s name of entity must be different from its singular form",
    "result": "error",
  },
]
`;
